hooks/tk-multi-publish2/basic/nuke_update_flame_clip.py

```diff
@@ -126,66 +131,125 @@ def accept(self, settings, item):
+        elif item.context.entity:
+            self.logger.debug("Attempting to find OpenClip publish in %s..." % item.context)
+
+            # TODO: We have a problem. The publish2 plugin acceptance routines
+            # are not re-run when the item's link changes. This means if the
+            # context hasn't been set to a shot using the panel app before
+            # publish2 is launched, we end up running this in the project
+            # context, which is never going to work for us here. Because we
+            # then aren't re-run if the user change's the item's link to be
+            # a Shot, we don't look for a clip file when we should have.
+            #
+            # As a fallback to the traditional approach of looking up the clip
+            # path using templates, we now look for a matching PublishedFile
+            # entity. This situation arises when dealing with publishes coming
+            # from Flame 2019+, which makes use of the publish2 app instead of
+            # the old flame export app.
+            publish_type = get_published_file_entity_type(publisher.sgtk)
+
+            # TODO: Should we get and update all of the clips that might be
+            # published? We're told this is an unlikely workflow by the Flame
+            # team, but it's technically possible. It raises other concerns,
+            # though, because then we have to ask the question of which (or all?)
+            # clips should be updated, and how we'd know that programmatically.
+            #
+            # UPDATE: We're going to do a middle-ground approach. If there are
+            # multiple clips, we're going to iterate over each one until we find
+            # one that has a local_path that exists. Once we find that, we use it.
+            # In most cases this will not do anything differently than the template
+            # driven solution, because we'll have a single OpenClip published for
+            # the shot, but there might be cases where this heads off a problem.
+            clip_publish_types = ["Flame OpenClip", "Flame Batch OpenClip"]
+            self.logger.debug("Clip publish types to search for: %s" % clip_publish_types)
+            clip_publishes = publisher.shotgun.find(
+                publish_type,
+                [
+                    ["entity", "is", item.context.entity],
+                    ["published_file_type.PublishedFileType.code", "in", clip_publish_types],
+                ],
+                fields=("path",),
             )
-            return {"accepted": False}
 
-        # ---- Ensure the clip path exists
+            if clip_publishes:
+                self.logger.debug("Found clip(s): %s" % clip_publishes)
+
+            for clip_publish in clip_publishes:
+                self.logger.debug("Checking existence of OpenClip: %s" % clip_publish)
+                try:
+                    flame_clip_path = resolve_publish_path(publisher.sgtk, clip_publish)
```

Comment:
@manneohrstrom Once I've found a PublishedFile that we're going to use, I can record that in the publish item for later use when we want to "version up" by creating a new PublishedFile. Do we have some reusable code lying around that will make creating the next available version of a given PublishedFile easier than coding it all out myself in this plugin?

Reply:
Can you just grab it from the `PublishedFile.version_number` and increment it? it shouldn't be a lot of code i would have thought!

```python
# figure out next version number
version_number = 1
if most_recent_publish:
   # there is already a publish in sg
   version_number = most_recent_publish["version_number"] + 1

# create new pub
register_publish(....)
```